service/collector/app.go

```diff
@@ -233,13 +234,29 @@ func (a *App) getTeam(ctx context.Context, app v1alpha1.App) (string, error) {
 		}
 	}
 
-	team = key.AppCatalogEntryTeam(*ace)
-	if team == "" {
-		// If there is no team annotation we use the default.
-		team = a.defaultTeam
+	return key.AppCatalogEntryTeam(*ace), nil
+}
+
+// getTeamMappings returns a map of AppCatalogEntry CR names to teams. This
+// reduces the number of API calls we need to make to fetch the teams metadata.
+func (a *App) getTeamMappings(ctx context.Context, apps []v1alpha1.App) (map[string]string, error) {
+	var teamMappings map[string]string
+
+	for _, app := range apps {
+		appCatalogEntryName := key.AppCatalogEntryName(key.CatalogName(app), key.AppName(app), key.Version(app))
```

Comment:
If we search appCatalogEntryName with app's name, app's catalog name and, app's version altogether, I think there are only a few cases you can reduce the number of calling k8s API. What do you having the app's name as key? I think we will hardly change the assigned team in the app after all.

Reply:
Yes for MCs apps in the control-plane catalog I don't think it will help.

My thought was it will help for WC apps in viking.

```
k get apps -A | wc -l
     587
k -n giantswarm get apps | wc -l
     70
```

But you're right a cache on app name would be better. We could also get the ACE CR with latest=true. I'll give that a try. Thx!

Reply:
> We could also get the ACE CR with latest=true. I'll give that a try. Thx!

That's a good idea!

Reply:
It doesn't work as listing the ACE CRs with labels is too slow :( 

Going to go back to the previous approach to fix the problem in viking. But fetching just once for each app would be better so will come back to this.